include/libhal-arm-mcu/stm32f1/timer.hpp

```diff
@@ -697,13 +673,35 @@ class advanced_timer final : public advanced_timer_manager
   {
     return advanced_timer_manager::acquire_pwm(static_cast<timer_pins>(p_pin));
   }
+  /**
+   * @brief Creates a quadrature encoder object used to measure the rotation of
+   * a motor that gives out quadrature encoder feedback.
+   *
+   * @param p_allocator Provide the resource that this can be allocated to.
+   * @param p_encoder_pins The pins must be part of the same timer and they must
+   * correspond to channel 1 and 2 respectively. Any channel other than channel
+   * 1 and 2 will throw an hal::operation_not_permitted error. If the channels
+   * are flipped, it will result degrees counted in the wrong direction.
+   * @param p_pulses_per_rotation Cycles per revolution for the quadrature
+   encoder.
 
-  [[nodiscard]] hal::stm32f1::quadrature_encoder acquire_quadrature_encoder(
-    encoder_pins p_encoder_pins)
+   * @return hal::v5::strong_ptr<hal::rotation_sensor>
+   *
+   * @throws hal::device_or_resource_busy - if timer is already being used with
+   * a conflicting resource.
+   * @throws hal::operation_not_permitted error - if any of the channels are not
+   * 1 and 2.
+   */
+  [[nodiscard]] hal::v5::strong_ptr<hal::rotation_sensor>
+  acquire_quadrature_encoder(std::pmr::polymorphic_allocator<> p_allocator,
+                             encoder_pins p_encoder_pins,
+                             float p_pulses_per_rotation)
   {
     return advanced_timer_manager::acquire_quadrature_encoder(
+      p_allocator,
       static_cast<timer_pins>(p_encoder_pins.channel_a_pin),
```

Comment:
```suggestion
      static_cast<timer_pins>(p_encoder_pins.channel_a),
```

Can you change the name to just channel_a. Adding the prefix of `_pin` isn't too useful as the class type is a pin so its already known by the reader that whats contained are pins.